pkg/graveler/staging/manager.go

```diff
@@ -82,7 +82,8 @@ func (m *Manager) Update(ctx context.Context, st graveler.StagingToken, key grav
 		oldValue = graveler.StagedEntryFromProto(oldValueProto)
 	}
 	updatedValue, err := updateFunc(oldValue)
-	if err != nil {
+	if err != nil || updatedValue == oldValue {
```

Comment:
`updatedValue == oldValue` is a bit subtle: it need to be the same slice, not just the same sequence of bytes.  I completely understand **why**, but I am worried that it will not be clear enough at the call site.

Some options:
* Improve documentation of this method in the interface.  Right now it says:

  ```go
	// Update updates a (possibly nil) value under the given staging token and key.
  ```
* Don't do it this way.  Instead add a new "error" return option for `updateFunc`, call it say `SkipUpdate` -- and document _that_ as something like
  > SkipUpdate is returned from ValueUpdateFunc to request _not_ to update the value.

Reply:
Using on special error